### DISCUSSION TITLE
Gemini seems to have a fake streaming expreience

### DISCUSSION THREAD

**Discussion by hjkcai** [danny-avila/LibreChat#3218] [Q&A]
### What happened?

When streaming of Gemini is enabled, the response seems to be returned in a constant speed. I have verified this through a self-written gemini proxy server. LibreChat always waits for the complete response from gemini then send to the user.

How to make it return the response in real-time? I can help submit a PR if I know how.

### Steps to Reproduce

Just use Gemini

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
This was done intentionally because the service returns **large** chunks of text at a time, which renders in an unpleasant way. I can look into how this behaves more recently but I was already planning on making stream output rates configurable and may look at that then.

**Comment by hjkcai** [comment]
> the service returns large chunks of text at a time

Yes. Gemini can sometimes return large chunks of text. However, I don't think the response stream should be buffered until the full response is received. This delay makes chatting with Gemini feel very slow. 

In my opinion, LibreChat could improve responsiveness by initially returning the original stream. Then, it could explore other ways to address the issue of large text chunks.

**Reply by R-Prady** [reply]
Can you share how you did the gemini proxy server? I’m interested in learning about this option and I’d your thoughts on why went this route? Thanks

**Reply by danny-avila** [reply]
@hjkcai I agree with you and I'll be making this behavior configurable
